### PR TITLE
[BugFix] Adjust NULL_TYPE hack processing(1)

### PR DESCRIPTION
**Pull request by liuyehcf**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes https://github.com/StarRocks/StarRocksTest/issues/1814

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Review by murphyatwork — DISMISSED** [review]

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 21 / 22 (95.45%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/ScalarOperatorToExpr.java|12|13|92.31%|[145]|
|:large_blue_circle:|com/starrocks/analysis/SlotRef.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/Expr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Type.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|1|1|100.00%|[]|

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by liuyehcf** [comment]
run starrocks_be_unittest

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 21 / 22 (95.45%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/ScalarOperatorToExpr.java|12|13|92.31%|[146]|
|:large_blue_circle:|com/starrocks/analysis/SlotRef.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/Expr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Type.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|1|1|100.00%|[]|

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 21 / 22 (95.45%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/ScalarOperatorToExpr.java|12|13|92.31%|[146]|
|:large_blue_circle:|com/starrocks/analysis/SlotRef.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/Expr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Type.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|1|1|100.00%|[]|

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 21 / 22 (95.45%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/ScalarOperatorToExpr.java|12|13|92.31%|[146]|
|:large_blue_circle:|com/starrocks/analysis/SlotRef.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/Expr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Type.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|1|1|100.00%|[]|

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 23 / 24 (95.83%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/ScalarOperatorToExpr.java|14|15|93.33%|[146]|
|:large_blue_circle:|com/starrocks/analysis/SlotRef.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/Expr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Type.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|1|1|100.00%|[]|

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 22 / 23 (95.65%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/ScalarOperatorToExpr.java|14|15|93.33%|[146]|
|:large_blue_circle:|com/starrocks/analysis/SlotRef.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/Expr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Type.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java [hunk 1]
```diff
@@ -316,7 +316,7 @@ public Void visitArrayExpr(ArrayExpr node, Scope scope) {
                     throw new SemanticException(e.getMessage());
                 }
             } else {
-                node.setType(new ArrayType(Type.NULL));
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java:319]: can you add some commet here which corresponds to the comment in ScalarOperatorToExpr

**Review by packy92 — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by Seaven — APPROVED** [review]

**Comment by jaogoy** [comment]
Why is there no any comment about this PR? including the reason of this bug, fixing details.

**Pull request merged**